pkg/renderer/html5/user_macro_test.go

```diff
@@ -0,0 +1,118 @@
+package html5_test
+
+import (
+	"html"
+	texttemplate "text/template"
+
+	"github.com/bytesparadise/libasciidoc/pkg/renderer"
+	. "github.com/onsi/ginkgo"
+)
+
+// an example for find and define user macro
+func findMacroTemplate() []renderer.Option {
+	t := texttemplate.New("user macro")
+	t.Funcs(texttemplate.FuncMap{
+		"escape": html.EscapeString,
+	})
+	tmpl := texttemplate.Must(t.ParseGlob("_macros/*"))
+	tmpls := tmpl.Templates()
+	opts := make([]renderer.Option, len(tmpls))
+	for i, tt := range tmpls {
+		opts[i] = renderer.DefineMacro(tt.Name(), tt)
+	}
+	return opts
+}
+
+var _ = Describe("user macros", func() {
+
+	Context("user macros", func() {
+		opts := findMacroTemplate()
+
+		It("undefined macro", func() {
+
+			actualContent := "hello::[]"
+			verifyError(GinkgoT(), actualContent)
```

Comment:
actually, if there is no such user-defined/custom `hello` macro, the output should be:

```
<div class="paragraph">
<p>hello::[]</p>
</div>
```

This is why I suggested that you also capture the "raw text" during the parsing, so it can be output in the HTML as-is.